app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/DummyDataSetInterface.kt

```diff
@@ -8,28 +13,51 @@ class DummyDatasetInterface : DatasetInterface {
     private val knife = DummyCategory("Knife")
     private val spoon = DummyCategory("Spoon")
 
-    val categories: Set<Category> = setOf(fork, knife, spoon)
+    val categories: MutableSet<Category> = mutableSetOf(fork, knife, spoon)
 
     private val forkPic = DummyCategorizedPicture(fork)
     private val knifePic = DummyCategorizedPicture(knife)
     private val spoonPic = DummyCategorizedPicture(spoon)
 
+    val pictures: MutableSet<CategorizedPicture> = mutableSetOf(forkPic, knifePic, spoonPic)
+
+
+
+
+    override suspend fun getPicture(category: Category): CategorizedPicture?{
+        if (!categories.contains(category)) throw IllegalArgumentException("The provided category" +
+                "is not present in the dataset")
+
+        for(p in pictures)
+            if(p == DummyCategorizedPicture(category)) return p
+
+        return null
+    }
+
+    override suspend fun putPicture(picture: Serializable, category: Category): CategorizedPicture {
```

Comment:
What do I do w/ a Serializable? this needs to be a more specific type, or a different one altogether.

Reply:
a Bitmap maybe? or a reference to one?